### ISSUE TITLE
deep learning/Bengio by deeppurple

### ISSUE THREAD

**Issue by bluejad** [bluejad/deeplearning#1]
luckly me

**Comment by bluejad** [comment]
![-3d97ee9f77ee5352](https://cloud.githubusercontent.com/assets/11881750/20409520/eeea5f00-ad54-11e6-8034-8d6aa9c14c6f.jpg)

**Comment by bluejad** [comment]
![img_20160806_143348_hdr](https://cloud.githubusercontent.com/assets/11881750/20409572/1fe8c3ee-ad55-11e6-85af-e757a2a99188.jpg)
![-30a8ebb33cccf99a](https://cloud.githubusercontent.com/assets/11881750/20409603/43ebe8a2-ad55-11e6-8602-21746b25ea4e.jpg)
![2015-02-16_103325](https://cloud.githubusercontent.com/assets/11881750/20409615/5628ff50-ad55-11e6-9dae-0c781b121b2d.jpg)

**Comment by bluejad** [comment]
![6810e892ad2c785e02bcea5989098b99](https://cloud.githubusercontent.com/assets/11881750/20419564/53e3b4fa-ad92-11e6-9a6c-722c2fb7171f.gif)
![721000640e580a2f1f8](https://cloud.githubusercontent.com/assets/11881750/20419565/53e70f06-ad92-11e6-8cdf-79f5e24301c9.jpg)
![721000842dbabb8fb42](https://cloud.githubusercontent.com/assets/11881750/20419569/53fea2e2-ad92-11e6-9817-f722fa26c646.jpg)
![20160111220953](https://cloud.githubusercontent.com/assets/11881750/20419568/53fa9396-ad92-11e6-8d1c-4df41d9029e4.jpeg)
![a6be614c9755871c325814ef74791d00_r](https://cloud.githubusercontent.com/assets/11881750/20419567/53fa7b2c-ad92-11e6-8c44-5321622ab589.jpg)

**Comment by bluejad** [comment]
Photo is beautiful!

**Comment by bluejad** [comment]
![expectation of empirical distribution pdata](https://cloud.githubusercontent.com/assets/11881750/20442064/19f3364c-ae02-11e6-8e15-b00132201cd1.png)
![maximum likelihood estimation 1fr](https://cloud.githubusercontent.com/assets/11881750/20442063/19f16e66-ae02-11e6-97e0-94d6873aaa43.png)
![maximum likelihood estimation 2sc](https://cloud.githubusercontent.com/assets/11881750/20442065/1a4e3fc4-ae02-11e6-8d2f-9ae3ae20a7c2.png)

**Comment by bluejad** [comment]
maximum likelihood estimation

![maximum likelihood estimation 1fr](https://cloud.githubusercontent.com/assets/11881750/20442188/a4049510-ae02-11e6-956c-4b1d49b44e66.png)

transfer into sum

![maximum likelihood estimation 2sc](https://cloud.githubusercontent.com/assets/11881750/20442187/a402891e-ae02-11e6-86da-de7c4d74f1a8.png)

the empirical distribution p^data by defined the training data

![expectation of empirical distribution pdata](https://cloud.githubusercontent.com/assets/11881750/20442186/a3dc1072-ae02-11e6-953b-4eb8f596d9dc.png)

**Comment by bluejad** [comment]
KL divergence

![kl divergenc](https://cloud.githubusercontent.com/assets/11881750/20443572/536d59ce-ae08-11e6-98dc-87276871cdb9.png)

**Comment by bluejad** [comment]
This means when we train the model to minimize the KL divergence, we need only minimize

![for minimum kl only need minimum this pic](https://cloud.githubusercontent.com/assets/11881750/20444024/5e09dedc-ae0a-11e6-83ea-01c283283ac2.png)

**Comment by bluejad** [comment]
the conditional maximum likelihood estimator 1fr

![the conditional maximum likelihood estimator 1fr](https://cloud.githubusercontent.com/assets/11881750/20454389/ce505e58-ae7a-11e6-8c98-e6a4581bec24.png)

the conditional maximum likelihood estimator 2sc

![the conditional maximum likelihood estimator 2sc](https://cloud.githubusercontent.com/assets/11881750/20454396/e7efcdbc-ae7a-11e6-8766-27f21f7a1a3b.png)

**Comment by bluejad** [comment]
the conditional distribution

![the conditioal distribution](https://cloud.githubusercontent.com/assets/11881750/20454523/43ae0b02-ae7e-11e6-902d-bf623cb41fb1.png)

**Comment by bluejad** [comment]
the conditional log-likel;ihood

![the conditional log-likelihood](https://cloud.githubusercontent.com/assets/11881750/20454581/d943f626-ae7f-11e6-9890-21c9e0f07dc8.png)

**Comment by bluejad** [comment]
解释最大似然估计的一种方法是将其视为最小化由训练集定义的经验分布pdata和模型分布之间的不相似性，两者之间的不相似度由KL散度测量

KL散度 KL divergence

![kl divergenc](https://cloud.githubusercontent.com/assets/11881750/20454664/3d04e5c4-ae82-11e6-86dc-338bb6a16dad.png)

**Comment by bluejad** [comment]
Any loss consisting of a negative log-likelihood is a cross entropy between the empirical distribution deﬁned by the training set and the model

任何由负对数似然组成的损失是由训练集和模型定义的经验分布之间的交叉熵

**Comment by bluejad** [comment]
the true data generating distribution pdata mean squared error is the cross-entropy between the empirical distribution and a Gaussian model

均方误差是经验分布和高斯模型之间的交叉熵

**Comment by bluejad** [comment]
the true data generating distribution pdata

真实数据生成分布pdata

**Comment by bluejad** [comment]
**p^(x；w)** 

the prediction of the mean of the Gaussian 

高斯的均值预测

**Comment by bluejad** [comment]
Comparing the log-likelihood with the mean squared error 

![comparing the log-likelihood with the mean squared error](https://cloud.githubusercontent.com/assets/11881750/20454691/33540860-ae83-11e6-855a-f362f8a3bb2c.png)

**Comment by bluejad** [comment]
Properties of Maximum Likelihood
- The true distribution pdata must lie within the model family pmodel(·; θ). Otherwise, no estimator can recover pdata.
- The true distribution pdata must correspond to exactly one value of θ. Otherwise, maximum likelihood can recover the correct pdata, but will not be able to determine which value of θ was used by the data generating processing.

**Comment by bluejad** [comment]
**p(θ)**

priorprobability distribution 

先验概率分布

**Comment by bluejad** [comment]
the prior via Bayes' rule

![the prior via bayes rule](https://cloud.githubusercontent.com/assets/11881750/20455001/51710fb6-ae8b-11e6-9e92-cc7e1c5d3db5.png)

**Comment by bluejad** [comment]
the predicted distribution of next data

![the predicted distribution of next data](https://cloud.githubusercontent.com/assets/11881750/20455037/38a9fb54-ae8c-11e6-97ad-ee3771773d0d.png)

**Comment by bluejad** [comment]
The prediction is parametrized by the vector w ∈ Rn

![the prediction is parametrized by the vector w rn](https://cloud.githubusercontent.com/assets/11881750/20461339/d689fdf2-af35-11e6-8d72-abb49a820bf8.png)

**Comment by bluejad** [comment]
Given a set of m training samples(X(train),y(train)), we can express the prediction of y over the entire training set as

![given a set of m training samples x train y train we can express the predictionof y over the entire training set as](https://cloud.githubusercontent.com/assets/11881750/20461350/2a315356-af36-11e6-88dc-82e93210556e.png)

**Comment by bluejad** [comment]
Expressed as a Gaussian conditional distribution on y(train), we have

![expressed as a gaussian conditional distribution on y train](https://cloud.githubusercontent.com/assets/11881750/20461361/73b9ca6c-af36-11e6-8530-e5805685bb51.png)

**Comment by bluejad** [comment]
For real-valued parameters it is common to use a Gaussian as a prior distribution

![for real-valued parameters it is common to use a gaussian as a prior distribution](https://cloud.githubusercontent.com/assets/11881750/20461366/ab8bd46c-af36-11e6-8675-564d7844a5cd.png)

**Comment by bluejad** [comment]
**Maximum A Posteriori (MAP) Estimation**

**Comment by bluejad** [comment]
The MAP estimate chooses the point of maximal  posterior probability (or maximal probability density in the more common case of continuous θ)

![the map estimate chooses the point of maximal posterior probability or maximal probability density in the more common case of continuous](https://cloud.githubusercontent.com/assets/11881750/20461380/0b8733a2-af37-11e6-9963-de6666df812e.png)

We recognize, above on the right hand side, logp(x | θ), i.e.the standard log-likelihood term, and log p(θ), corresponding to the prior distribution

**Comment by bluejad** [comment]
**Probabilistic Supervised Learning**

Most supervised learning algorithms in this book are based on estimating a probability distribution **p(y | x)**

We can do this simply by using maximum likelihood estimation to ﬁnd the best parameter vector θ for a parametric family of distributions **p(y | x;θ)**

We have already seen that linear regression corresponds to the family

![we have already seen that linear regression corresponds to the family](https://cloud.githubusercontent.com/assets/11881750/20461399/188da576-af38-11e6-9390-9e01cfdfd196.png)

One way to solve this problem is to use the logistic sigmoid function to squash the output of the linear function into the interval (0, 1) and interpret that value as a probability

![one way to solve this problem is to use](https://cloud.githubusercontent.com/assets/11881750/20461412/6480fd5c-af38-11e6-8854-8e5244da6e03.png)

**Comment by bluejad** [comment]
negative log-likelihood (NLL)

**Comment by bluejad** [comment]
negative log-likelihood (NLL)

**Comment by bluejad** [comment]
**Support Vector Machines**

This model is similar to logistic regression in that it is driven by a linear function 

![support vector machines](https://cloud.githubusercontent.com/assets/11881750/20461428/e28c972e-af38-11e6-94db-82e9d2ffc00f.png)

The SVM predicts that the positive class is present when **wTx +b** is positive. Likewise, it predicts that the negative class is present when **wTx+b** is negative

One key innovation associated with support vector machines is the kernel trick

![one key innovation associated with support vector machines is the kernel trick](https://cloud.githubusercontent.com/assets/11881750/20461438/393603e4-af39-11e6-87d0-06e3362c79dd.png)

Rewriting the learning algorithm this way allows us to replace **x** by the output of a given feature function **φ(x)** and the dot product with a function **k(x,x(i)) = φ(x)·φ(x(i))** called a kernel

以这种方式重写学习算法允​​许我们用给定特征函数**φ（x）**的输出和具有函数
**k（x，x（i））=φ（x）·φ（x（i） ））**称为内核

After replacing dot products with kernel evaluations, we can make predictions using the function

![after replacing dot products with kernel evaluations we can make predictions using the function](https://cloud.githubusercontent.com/assets/11881750/20461480/962115ac-af3a-11e6-81e1-b1a6e67f52aa.png)

**Comment by bluejad** [comment]
**Other Simple Supervised Learning Algorithms**

![other simple supervised learning algorithms 1fr](https://cloud.githubusercontent.com/assets/11881750/20461500/1c14c06e-af3b-11e6-9c1a-0624913aea88.png)

![other simple supervised learning algorithms 2sc](https://cloud.githubusercontent.com/assets/11881750/20461501/217cca6a-af3b-11e6-961d-1189c9d0678d.png)

**Comment by bluejad** [comment]
**Unsupervised Learning Algorithms**

The unbiased sample covariance matrix associated with X is given by

与X相关的无偏样本协方差矩阵由下式给出

![the unbiased sample covariance matrix associated with x is given by](https://cloud.githubusercontent.com/assets/11881750/20461523/d3e15252-af3b-11e6-93b6-6e85e219a9bf.png)

PCA ﬁnds a representation (through linear transformation) **z = xTW** where **Var[z]** is diagonal

PCA表示（通过线性变换）**z = xT W**，其中**Var [z]**是对角线

we saw that the principal components of a design matrix **X** are given by the eigenvectors of **XTX**. From this view

![let w be theright singular vectors in the decomposition x u w](https://cloud.githubusercontent.com/assets/11881750/20461554/0131e720-af3d-11e6-8720-7b567545ac98.png)

**WWT = I**

**Comment by bluejad** [comment]
**Stochastic Gradient Descent**

the negative conditional log-likelihood of the training data can be written as

![the negative conditional log-likelihood of the training data can be written as](https://cloud.githubusercontent.com/assets/11881750/20461602/df7efcb6-af3d-11e6-9180-236f3ffab138.png)

The computational cost of this operation is O(m)

The estimate of the gradient is formed as

![the estimate of the gradient is formed as](https://cloud.githubusercontent.com/assets/11881750/20461609/2304ed06-af3e-11e6-9730-85a523b37ecd.png)

using examples from the minibatch B. The stochastic gradient descent algorithm then follows the estimated gradient downhill:

![the stochastic gradient descent algorithm s is the learning rate](https://cloud.githubusercontent.com/assets/11881750/20461626/7f0da142-af3e-11e6-886d-300a59cabcd6.png)

learning rate

![learning rate](https://cloud.githubusercontent.com/assets/11881750/20461802/11cbbca8-af44-11e6-89a1-b2c3ae3e5a46.png)

**Comment by bluejad** [comment]
**Building a Machine Learning Algorithm**

the linear regression algorithm combines a dataset consisting of **X** and **y**, the cost function

![the linear regression algorithm combines a dataset consisting of x and y the cost function pmodel y x n y xw b 1](https://cloud.githubusercontent.com/assets/11881750/20461685/5e2138c0-af40-11e6-82db-bdde5f96d630.png)

the model speciﬁcation pmodel(y | **x**) = N (y;**xTw**\+ b,**1**)

in most cases, the optimization algorithm deﬁned by solving for where the gradient of the cost is zero using the normal equations

优化算法是通过求解成本的梯度为零

we can add weight decay to the linear regression cost function to obtain

![we can add weight decay to the linear regression cost function to obtain](https://cloud.githubusercontent.com/assets/11881750/20461711/13ee591c-af41-11e6-8852-27c49ae1d20c.png)

This still allows closed-form optimization

we can obtain the ﬁrst PCA vector by specifying that our loss function is

![we can add weight decay to the linear regression cost function to obtain](https://cloud.githubusercontent.com/assets/11881750/20461722/4dce301c-af41-11e6-8709-a68f456f830c.png)

while our model is deﬁned to have w with norm one and reconstruction function **r(x) = wTxw**

而我们的模型被定义为具有范数w和重建函数**r（x）= wT xw**

**Comment by bluejad** [comment]
**Challenges Motivating Deep Learning**

As the number of relevant dimensions of the data increases (from left to right), the number of conﬁgurations of interest may grow exponentially

![as the number of relevant dimensions of the data increases from left to](https://cloud.githubusercontent.com/assets/11881750/20461735/ef1141ee-af41-11e6-8f29-0ee415334f30.png)

Illustration of how the nearest neighbor algorithm breaks up the input space into regions

最近邻算法如何将输入空间分解为区域的图示

![illustration of how the nearest neighbor algorithm breaks up the input space](https://cloud.githubusercontent.com/assets/11881750/20461742/25dd09ce-af42-11e6-9be8-88c3f6cb7031.png)

**Comment by bluejad** [comment]
**Part II Deep Networks: Modern Practices**

**Comment by bluejad** [comment]
**multilayer perceptrons (MLPs)**

**Comment by bluejad** [comment]
A feedforward network deﬁnes a mapping **y = f(x；θ)** and learns the value of the parameters θ that result in the best function approximation

前馈网络定义映射**y = f（x；θ）**，并且学习导致最佳函数近似的参数θ的值

**Comment by bluejad** [comment]
**no feedback connections**

**Deep feedforward networks**   

**feedforward neural networks**   

**multilayer perceptrons (MLPs)**

**Comment by bluejad** [comment]
**feedback connections**

**recurrent neural networks**

**Comment by bluejad** [comment]
The goal of a feedforward network is to approximate some function f∗

前馈网络的目的是近似某个函数f *

**Comment by bluejad** [comment]
The dimensionality of these hidden layers determines the width of the model

这些隐藏层的维度决定了模型的宽度

**Comment by bluejad** [comment]
**Example: Learning XOR**

The XOR function (“exclusive or”) is an operation on two binary values, x1 and x2. When exactly one of these binary values is equal to 1, the XOR function returns 1. Otherwise, it returns 0. The XOR function provides the target function y = f_(x) that we want to learn. Our model provides a function y =f(x；θ) and ourlearning algorithm will adapt the parameters θ to make f assimilar as possible to f_

Evaluated on our whole training set, the MSE loss function is

![evaluated on our whole training set the mse loss function is](https://cloud.githubusercontent.com/assets/11881750/20462977/56f173d4-af64-11e6-9339-7ca7995262c3.png)

Now we must choose the form of our model, **f(x；θ)**. Suppose that we choose a linear model, with **θ** consisting of **w** and b. Our model is deﬁned to be
f(**x；w**, b) = **xTw** + b

minimize J(**θ**)

four points X = {[0,0]T, [0,1]T,[1,0]T, and [1,1]T}

Answer **w** = 0 and b = 1/2
